internal/services/aws_iam_service.go

```diff
@@ -15,6 +17,10 @@ import (
 func ValidatePermissions(w http.ResponseWriter, r *http.Request) {
 	logger := zerolog.Ctx(r.Context())
 	sourceId := chi.URLParam(r, "ID")
+	if err := middleware.DigitsOnly(sourceId); err != nil {
+		renderError(w, r, payloads.NewURLParsingError(r.Context(), "id parameter invalid", err))
+	}
+
 	region := r.URL.Query().Get("region")
 
 	if region == "" {
```

Comment:
I highly suggest to avoid creating cyclic dependencies, what if middleware wants to import this package? Then we have a problem. I suggest to find a new home for the validation function and if there is no such place simply create a brand new `validate` or similar package. It might sound weird, but this is going to pay off in the future.

You can take advantage of the fact that there is no link between parent and child package in go, so you can go ahead and create `payload/validate` package and it will be a completely separate entity but still you will not "pollute" the root folder.

Reply:
Let me know if you feel better about the structure now:)